kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -216,6 +216,12 @@ public String createConsumer(String group, ConsumerInstanceConfig instanceConfig
       // and should not be propagated to the consumer
       props.setProperty("request.timeout.ms", "30000");
 
+      // If the consumer.timeout.ms value is set higher than the default value
+      // for max.poll.interval.ms then it is possible the consumer will be
+      // considered failed by the brokers while it has yet to hit the timeout
+      // for the rest proxy.
+      props.setProperty("max.poll.interval.ms", props.getProperty("consumer.timeout.ms", ""));
```

Comment:
The consumer should not be considered failed by the brokers if it violates the `max.poll.interval.ms`. The validation is done in the client side (here: https://github.com/apache/kafka/blob/0d55f0f3ec8f97bc250b325481f6f2fa70f52a5c/clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java#L1052)

In the case of the v2 consumer, `max.poll.interval.ms` should be close to `request.timeout.ms`. For what it's worth, I don't think we should set this property in the code for the v2 consumer at all. Users could configure it in their proxy config via adding the `consumer.` prefix to it.

Reply:
Additionally, `consumer.timeout.ms` is not something we have in any config nor is it documented anywhere in the proxy. It is part of the old consumer configs (https://kafka.apache.org/20/documentation.html#oldconsumerconfigs)